permits/api.py

```diff
@@ -0,0 +1,186 @@
+        2.- /rest/permits/?works-object-type=1&status=0
+        3.- /rest/permits/?geom-type=lines | points | polygons
+    """
+
+    serializer_class = serializers.PermitRequestPrintSerializer
+
+    def get_queryset(self):
+        """
+        This view should return a list of permits for which the logged user has
+        view permissions
+        """
+        user = self.request.user
+
+        if not user.is_authenticated:
+            raise APIException(services.EndpointErrors.NO_AUTH.value)
+
+        work_objects_type = self.request.query_params.get("works-object-type", None)
+        status = self.request.query_params.get("status", None)
+        geom_type = self.request.query_params.get("geom-type", None)
+        permitrequest_id = self.request.query_params.get("permit-request-id", None)
+
+        base_filter = Q()
+
+        if work_objects_type:
+            if work_objects_type.isdigit():
+                base_filter &= Q(works_object_types=work_objects_type)
+            else:
+                raise APIException(services.EndpointErrors.WOT_NOT_INT.value)
+
+        if status:
+            if status.isdigit():
+                base_filter &= Q(status=status)
+            else:
+                raise APIException(services.EndpointErrors.STATUS_NOT_INT.value)
+
+        if permitrequest_id:
+            if permitrequest_id.isdigit():
+                base_filter &= Q(pk=permitrequest_id)
+            else:
+                raise APIException(services.EndpointErrors.PR_NOT_INT.value)
+
+        works_object_types_prefetch = Prefetch(
+            "works_object_types",
+            queryset=models.WorksObjectType.objects.select_related("works_type"),
+        )
+
+        geom_qs = models.PermitRequestGeoTime.objects.only("geom")
+
+        if geom_type:
+            if geom_type not in ("lines", "points", "polygons"):
+                raise APIException(services.EndpointErrors.GEO_NOT_VALID.value)
+            geom_qs = geom_qs.annotate(geom_type=GeomToText(F("geom"),))
+            if geom_type == "lines":
+                geom_qs = geom_qs.filter(geom_type__icontains="line")
+            if geom_type == "points":
+                geom_qs = geom_qs.filter(geom_type__icontains="point")
+            if geom_type == "polygons":
+                geom_qs = geom_qs.filter(geom_type__icontains="poly")
+            base_filter &= Q(id__in=geom_qs)
+
+        geotime_prefetch = Prefetch("geo_time", queryset=geom_qs)
+
+        try:
+            qs = (
+                models.PermitRequest.objects.filter(base_filter)
+                .filter(
+                    Q(id__in=services.get_permit_requests_list_for_user(user))
+                    | Q(is_public=True)
+                )
+                .prefetch_related(works_object_types_prefetch)
+                .prefetch_related(geotime_prefetch)
+                .select_related("administrative_entity")
+            )
+        except ValueError as e:
```

Comment:
Je vois pas ce qui pourrait renvoyer une `ValueError` ici?

Reply:
Not anymore, previous geom_filtering generated that... Removed!